### ISSUE TITLE
Update Fenix PR code to account for new location

### ISSUE THREAD

**Issue by rvandermeulen** [mozilla/application-services#5383]
As of yesterday, Fenix now lives in the firefox-android monorepo alongside AC and Focus. The logic for scheduling those builds and tests on PRs needs to be updated to take that into account.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-3607)

**Comment by rvandermeulen** [comment]
Or maybe we can just remove the Fenix code? I see `branch-build-fenix-build` being scheduled on a PR with just `[ff-android: main]` specified.

**Comment by rvandermeulen** [comment]
Huh, so those Fenix jobs appear to be coming from the old repo (which are now perma-broken due to the most recent A-S bump). I'm also not sure why they got scheduled in the first place when I didn't specify Fenix builds in my PR title. Any ideas, @bendk?

**Comment by bendk** [comment]
I think we have a nightly cron job to run them, maybe that's where they're coming from.

I'm hoping to just rip out this code as part of the rapid-release migration rather than fix it.  Would it cause problems in the meantime?  I expect to have something merged in the next couple weeks.

**Comment by rvandermeulen** [comment]
It makes it difficult to test PRs touching Fenix-adjacent components because the builds are perma-busted.

**Comment by rvandermeulen** [comment]
FWIW, I think it's still useful to be able to test a PR against AC/Fenix before it ships in a nightly A-S build.

**Comment by bendk** [comment]
> It makes it difficult to test PRs touching Fenix-adjacent components because the builds are perma-busted.

Good point, I made #5406.  Hopefully it fixes things.

> FWIW, I think it's still useful to be able to test a PR against AC/Fenix before it ships in a nightly A-S build.

Agreed.  I think the new way to handle this will be to:
 - Create an application-services try-style build.  This will build app-services packages, put them in a separate  with a version string like `111.0.my-feature-branch`.  The packages should probably live in a separate place, maybe alongside nightlies or maybe in their own directory.  I think ship-it can help automate this.
 - Open a `do-not-land` firefox-android PR that changes the version string to run the CI.  This PR is also where you would resolve any breaking changes.
 - If there are breaking changes, then when the `application-services` branch gets merged and released in a nightly, then the nightly `firefox-android` PR will fail.  To fix this, merge the `firefox-android` branch for the feature branch into the branch for the nightly PR.

**Comment by rvandermeulen** [comment]
Yeah, that'll work, I guess. Just not quite as turnkey as the PR title solution was previously. And it'll also require the submitter to have Ship-It access, which effectively limits it to employees only.

**Comment by bendk** [comment]
Maybe we could keep both systems?

If there are breaking changes that need a firefox-android PR, I think it's nicer to have the CI run in that PR -- especially since we want the android devs to approve the changes before we merge the corresponding changes into application-services.

But if it's just about changing Fenix-adjacent components that live in application-services, then the branch-build system seems simpler.

One concern I have is the long-term maintenance.  When I set things up initially, I had to tweak a lot of small things in order to get the tests to pass.  It seems like that might be needed in the future as the android devs change their tests.  But that hasn't been an issue yet and hopefully we will eventually get to live with the Android code in a monorepo, so maybe this won't be an issue.
